auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -469,25 +546,265 @@ public Builder(@SupportedProvider @NonNull String providerId) {
+
+                getParams().putString(ExtraConstants.EXTRA_COUNTRY_ISO, iso);
+                getParams().putString(ExtraConstants.EXTRA_NATIONAL_NUMBER, number);
+
+                return this;
+            }
+
+            /**
+             * Set the default country code that will be used in the phone verification sign-in
+             * flow.
+             *
+             * @param iso country iso
+             */
+            @NonNull
+            public PhoneBuilder setDefaultCountryIso(@NonNull String iso) {
+                if (getParams().containsKey(ExtraConstants.EXTRA_PHONE)
+                        || getParams().containsKey(ExtraConstants.EXTRA_COUNTRY_ISO)
+                        || getParams().containsKey(ExtraConstants.EXTRA_NATIONAL_NUMBER)) {
+                    throw new IllegalStateException("Cannot overwrite previously set phone number");
+                }
+
+                getParams().putString(ExtraConstants.EXTRA_COUNTRY_ISO, iso);
+
+                return this;
+            }
+        }
+
+        /**
+         * {@link IdpConfig} builder for the Google provider.
+         */
+        public static final class GoogleBuilder extends Builder {
+            public GoogleBuilder() {
+                //noinspection deprecation taking a hit for the backcompat team
+                super(GoogleAuthProvider.PROVIDER_ID);
+                if (getApplicationContext().getString(R.string.default_web_client_id)
+                        .equals(UNCONFIGURED_CONFIG_VALUE)) {
+                    throw new RuntimeException("Check your google-services plugin configuration," +
+                            " the client id string wasn't populated.");
+                }
+            }
+
+            /**
+             * Set the scopes that your app will request when using Google sign-in. See all <a
+             * href="https://developers.google.com/identity/protocols/googlescopes">available
+             * scopes</a>.
+             *
+             * @param scopes additional scopes to be requested
+             */
+            @NonNull
+            public GoogleBuilder setScopes(@NonNull List<String> scopes) {
+                GoogleSignInOptions.Builder builder =
+                        new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN);
+                for (String scope : scopes) {
+                    builder.requestScopes(new Scope(scope));
+                }
+                return setSignInOptions(builder.build());
+            }
+
+            /**
+             * Set the {@link GoogleSignInOptions} to be used for Google sign-in. Standard options
+             * like requesting the user's email will automatically be added.
+             *
+             * @param options sign-in options
+             */
+            @NonNull
+            public GoogleBuilder setSignInOptions(@NonNull GoogleSignInOptions options) {
+                if (getParams().containsKey(ExtraConstants.EXTRA_GOOGLE_SIGN_IN_OPTIONS)) {
```

Comment:
nit: Seems like we are doing checks of this style a lot, could be refactored into a `checkUnset(key, errorMsg)` method.

Reply:
Done.